internal/rest/avi_evh_obj.go

```diff
@@ -492,6 +501,10 @@ func (rest *RestOperations) AviVsChildEvhBuild(vs_meta *nodes.AviEvhVsNode, rest
 		VhType:                proto.String(utils.VS_TYPE_VH_ENHANCED),
 	}
 
+	if len(vs_meta.ICAPProfileRefs) != 0 {
+		evhChild.IcapRequestProfileRefs = &vs_meta.ICAPProfileRefs
```

Comment:
Do we have to attach the ICAP profiles to SNI Child VS also?

Reply:
Working on implementing ICAP for SNI mode also

Reply:
Have added code for adding ICAP profile for SNI child VS